src/gccrs/args.rs

```diff
@@ -0,0 +1,153 @@
+//! This module interprets arguments given to `rustc` and transforms them into valid
+//! arguments for `gccrs`.
+
+use getopts::{Matches, Options};
+
+/// Crate types supported by `gccrs`
+#[derive(Clone, Copy)]
+pub enum CrateType {
+    /// Binary application
+    Bin,
+    /// Dynamic library/Shared object
+    DyLib,
+    /// Statically linked library
+    StaticLib,
+    /// Remaining options, handled by `rustc` but not `gccrs`
+    Unknown,
+}
+
+impl CrateType {
+    /// Get the corresponding [`CrateType`] from a given option to the `--crate-type`
+    /// option
+    pub fn from_str(s: &str) -> CrateType {
+        match s {
+            "bin" => CrateType::Bin,
+            // FIXME: Is 'lib' really part of this? Not always, right?
+            "dylib" | "lib" => CrateType::DyLib,
```

Comment:
lib is an alias for rlib atm. it is officially documented as changable to dylib in the future, but I would consider it highly unlikely because it would break things.

Reply:
Thanks, will fix!